File: be/src/users/dto/user-profile.dto.ts
import { Expose } from 'class-transformer';

export class UserProfileDto {
  @Expose() id: number;
  @Expose() name: string;
  @Expose() email: string;
  @Expose() avatar: string;
  @Expose() studentCode: string;
  @Expose() department: string;
  @Expose() createdAt: Date;
}

File: backend-java/src/main/java/com/qlda/backend/users/dto/UserProfileDto.java
package com.qlda.backend.users.dto;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
public class UserProfileDto {
    private String id;
    private String name;
    private String email;
    private String avatar;
    private String studentCode;
    private String department;
    private LocalDate dateOfBirth;
    private String gender;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

